working/333 - shared memory multithreading/proposal.md

```diff
@@ -1096,43 +793,6 @@ All built-in asynchronous primitives will make the following API guarantee: **a
 callback passed to `Future` or `Stream` APIs will be invoked using executor
 which was running the code which registered the callback.**
```

Comment:
Can you share futures or streams at all? (They're not trivally shareable, so likely not in the initial proposal.)

If you can, then we'll have solved the problem of `Zone.current` being non-shared state, and every operation on `Future` accessing `Zone.current`. That's not going to be trivial.

(Maybe we do want some non-shared global variables that are made accessible, so we can initialize them if necesssary..)

Reply:
You can't share them in the SNMmt (_shared **native memory** multithreading_) but you can share them in SEmt (_shared **everything** multithreading_).